Undefined references in Windows CVODE FMU export

### Description

The FMU export is broken on Windows when trying to use the CVODE solver.

### Steps to Reproduce

Run the following script with `omc`
```modelica
loadModel(Modelica, {"4.0.0"}); getErrorString();
setCommandLineOptions("-d=newInst --fmiFlags=s:cvode"); getErrorString();
buildModelFMU(Modelica.Blocks.Examples.PID_Controller); getErrorString();
```

Output:
```
[...]
g++ -shared -I. -o Modelica_Blocks_Examples_PID_Controller.dll  simulation/solver/cvode_solver.o simulation/solver/sundials_error.o Modelica_Blocks_Examples_PID_Controller.o Modelica_Blocks_Examples_PID_Controller_functions.o Modelica_Blocks_Examples_PID_Controller_records.o Modelica_Blocks_Examples_PID_Controller_01exo.o Modelica_Blocks_Examples_PID_Controller_02nls.o Modelica_Blocks_Examples_PID_Controller_03lsy.o Modelica_Blocks_Examples_PID_Controller_04set.o Modelica_Blocks_Examples_PID_Controller_05evt.o Modelica_Blocks_Examples_PID_Controller_06inz.o Modelica_Blocks_Examples_PID_Controller_07dly.o Modelica_Blocks_Examples_PID_Controller_08bnd.o Modelica_Blocks_Examples_PID_Controller_09alg.o Modelica_Blocks_Examples_PID_Controller_10asr.o Modelica_Blocks_Examples_PID_Controller_11mix.o Modelica_Blocks_Examples_PID_Controller_12jac.o Modelica_Blocks_Examples_PID_Controller_13opt.o Modelica_Blocks_Examples_PID_Controller_14lnz.o Modelica_Blocks_Examples_PID_Controller_15syn.o Modelica_Blocks_Examples_PID_Controller_16dae.o Modelica_Blocks_Examples_PID_Controller_17inl.o Modelica_Blocks_Examples_PID_Controller_18spd.o Modelica_Blocks_Examples_PID_Controller_init_fmu.o Modelica_Blocks_Examples_PID_Controller_FMU.o -I\"C:/Program Files/OpenModelica1.19.0-dev-64bit/include/omc/c/fmi\" -DOMC_FMI_RUNTIME=1 -DWITH_SUNDIALS=1 -Isundials -DCMINPACK_NO_DLL=1 -L\"d:/workspace/OpenModelica/build/lib/omlibrary/Modelica 4.0.0/Blocks\"   -Os -Wno-parentheses-equality -falign-functions -mstackrealign -msse2 -mfpmath=sse  -L\"C:/Program Files/OpenModelica1.19.0-dev-64bit/lib//omc\" -Wl,-rpath,\"C:/Program Files/OpenModelica1.19.0-dev-64bit/lib//omc\" -fopenmp -Wl,-Bstatic -lregex -ltre -lintl -liconv -lexpat -lomcgc -lpthread -loleaut32 -limagehlp -lhdf5 -lz -lszip -Wl,-Bdynamic  -lSimulationRuntimeFMI  -Wl,-Bstatic -lregex -ltre -lintl -liconv -static-libgcc -lpthread -lm  -lgfortran -lquadmath -lmingw32 -lgcc_eh -lmoldname -lmingwex  -luser32 -lkernel32 -ladvapi32 -lshell32 -limagehlp -lopenblas -lhdf5 -lz -lszip -Wl,-Bdynamic  -Wl,-Bstatic -lsundials_cvode -lsundials_nvecserial -Wl,-Bdynamic -llis -Wl,--kill-at
C:/OMDev/tools/msys/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: simulation/solver/cvode_solver.o:cvode_solver.c:(.text+0x8b8): undefined reference to `__imp_N_VMake_Serial'
C:/OMDev/tools/msys/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: simulation/solver/cvode_solver.o:cvode_solver.c:(.text+0x8cd): undefined reference to `__imp_CVodeCreate'
C:/OMDev/tools/msys/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: simulation/solver/cvode_solver.o:cvode_solver.c:(.text+0x8f7): undefined reference to `__imp_CVodeInit'
C:/OMDev/tools/msys/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: simulation/solver/cvode_solver.o:cvode_solver.c:(.text+0x97d): undefined reference to `__imp_N_VMake_Serial'
C:/OMDev/tools/msys/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: simulation/solver/cvode_solver.o:cvode_solver.c:(.text+0x9a1): undefined reference to `__imp_CVodeSVtolerances'
[...]
collect2.exe: error: ld returned 1 exit status
make: *** [Makefile:60: nozip] Fehler 1
"
```

Here is the complete log: 
[build.log](https://github.com/OpenModelica/OpenModelica/files/7914707/build.log)


### Expected Behavior

Link CVODE into the FMU.

### Version and OS

<!--- Please complete the following information. -->

 - OpenModelica Version: v1.19.0-dev-541-gebaac9b11c
 - OS: Windows 10, 64 bit

--- a/OMCompiler/Compiler/Script/CevalScriptBackend.mo
+++ b/OMCompiler/Compiler/Script/CevalScriptBackend.mo
@@ -3693,7 +3693,7 @@ algorithm
   end if;
   if needs3rdPartyLibs then
     SUNDIALS :=  "1";
-    CPPFLAGS := CPPFLAGS + " -DWITH_SUNDIALS=1" + " -Isundials";
+    CPPFLAGS := CPPFLAGS + " -DWITH_SUNDIALS=1 -DLINK_SUNDIALS_STATIC" + " -Isundials";
   else
     SUNDIALS :=  "";
   end if;

